### PR TITLE
SCALRCORE-19846 > Publish Modules to preview API

### PR DESCRIPTION
**Pull request by penja**

### CODE REVIEW

File: module.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	Create(ctx context.Context, options ModuleCreateOptions) (*Module, error)
+	// Read a module by its ID.
+	Read(ctx context.Context, moduleID string) (*Module, error)
+	// ReadBySource Read the module by its Source.
+	ReadBySource(ctx context.Context, moduleSource string) (*Module, error)
+	// Delete a module by its ID.
+	Delete(ctx context.Context, moduleID string) error
+}
+
+// runs implements Runs.
```
Comment (lbn) [inline module.go:28]: Comment.

File: module.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	Account             *Account            `jsonapi:"relation,account,omitempty"`
+	Environment         *Environment        `jsonapi:"relation,environment,omitempty"`
+	CreatedBy           *User               `jsonapi:"relation,created-by,omitempty"`
+	LatestModuleVersion *ModuleVersion      `jsonapi:"relation,latest-module-version,omitempty"`
+}
+
+// ModuleStatus represents a module state.
+type ModuleStatus string
+
+//List all available run statuses.
```
Comment (lbn) [inline module.go:53]: Comment.

**Review by lbn — CHANGES_REQUESTED** [review]

File: module.go (hunk 2 above)
Comment (penja) [reply module.go:53]: @lbn ty

File: module.go (hunk 1 above)
Comment (penja) [reply module.go]: fixed

**Review by lbn — APPROVED** [review]

**Pull request merged**